extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/GraphiteEmitter.java

```diff
@@ -0,0 +1,180 @@
+        exec.scheduleAtFixedRate(
+            new ConsumerRunnable(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            graphiteEmitterConfig.getFlushPeriod(),
+            TimeUnit.MILLISECONDS
+        );
+        started.set(true);
+      }
+    }
+  }
+
+
+  @Override
+  public void emit(Event event)
+  {
+    if (!started.get()) {
+      throw new ISE("WTF emit was called while service is not started yet");
+    }
+
+    if (event instanceof ServiceMetricEvent) {
+      final ServiceMetricEvent serviceMetricEvent = (ServiceMetricEvent) event;
+      final GraphiteEvent graphiteEvent = graphiteEventConverter.druidMetricToGraphite(serviceMetricEvent);
+      if (graphiteEvent == null) {
+        return;
+      }
+      try {
+        final boolean isSuccessful = eventsQueue.offer(graphiteEvent, DEFAULT_PUT_GET_TIMEOUT, TimeUnit.MILLISECONDS);
+        if (!isSuccessful) {
+          log.error(
+              "Throwing event [%s] on the floor Graphite queue is full please increase the capacity or/and the consumer frequency",
+              mapper.writeValueAsString(event)
+          );
+        }
+      }
+      catch (InterruptedException | JsonProcessingException e) {
+        Thread.interrupted();
+        log.error(e, e.getMessage(), Thread.interrupted());
+      }
+    } else if (graphiteEmitterConfig.getLogAlerts() && event instanceof AlertEvent) {
+      try {
+        log.warn(mapper.writeValueAsString(event));
+      }
+      catch (JsonProcessingException e) {
+        log.error(e, "error while serializing alert event[%s]", ((AlertEvent) event).getDescription());
+      }
+    }
+  }
+
+  private class ConsumerRunnable implements Runnable
+  {
+    private PickledGraphite pickledGraphite = new PickledGraphite(
+        graphiteEmitterConfig.getHostname(),
+        graphiteEmitterConfig.getPort(),
+        graphiteEmitterConfig.getBatchSize()
+    );
+
+    @Override
+    public void run()
+    {
+      try {
+
+        if (!pickledGraphite.isConnected()) {
+          log.info("trying to connect to graphite server");
+          pickledGraphite.connect();
+        }
+        while (eventsQueue.size() > 0 && !exec.isShutdown()) {
+          try {
+            final GraphiteEvent graphiteEvent = eventsQueue.poll(DEFAULT_PUT_GET_TIMEOUT, TimeUnit.MILLISECONDS);
+            if (graphiteEvent != null) {
+              log.debug(
+                  "sent [%s] with value [%s] and time [%s]",
+                  graphiteEvent.getEventPath(),
+                  graphiteEvent.getValue(),
+                  graphiteEvent.getTimestamp()
+              );
+              pickledGraphite.send(
+                  graphiteEvent.getEventPath(),
+                  graphiteEvent.getValue(),
+                  graphiteEvent.getTimestamp()
+              );
+            }
+          }
+          catch (InterruptedException | IOException e) {
+            Thread.interrupted();
+            log.error(e, e.getMessage());
+          }
```

Comment:
i meant, "throw some exception" on receiving InterruptedException ... whoever interrupted this thread wanted the thread to finish asap.